client/state/jetpack-connect/actions.js

```diff
@@ -348,22 +348,22 @@ export default {
 				} );
 				// Update the user now that we are fully connected.
 				userFactory().fetch();
-				return wpcom.me().sites( { site_visibility: 'all', include_domain_only: true } );
+				return wpcom.site( parseInt( client_id ) ).get();
 			} )
 			.then( ( data ) => {
 				tracksEvent( dispatch, 'calypso_jpc_auth_sitesrefresh', {
 					site: client_id
 				} );
 				debug( 'Sites list updated!', data );
 				dispatch( {
-					type: SITES_RECEIVE,
-					sites: data.sites
+					type: SITE_RECEIVE,
+					site: omit( data, '_headers' )
 				} );
 				dispatch( {
-					type: JETPACK_CONNECT_AUTHORIZE_RECEIVE_SITE_LIST,
-					data: data
+					type: SITE_REQUEST_SUCCESS,
```

Comment:
To be consistent, we should also make sure to dispatch a `SITE_DELETE_FAILURE` if the request fails. Maybe we could directly inject `requestSite()` instead of duplicating its functionality?